modeling-data-relationships/modeling-answers.md

```diff
@@ -0,0 +1,121 @@
+1)
+
+a)
+
+```
+SELECT guests.first_name, guests.last_name
+FROM guests
+LEFT JOIN bookings ON bookings.guest_id = guests.id
+WHERE bookings.guest_id IS NULL;
+```
+b) 
+```
+SELECT b.check_in, b.check_out, g.first_name, g.last_name, COUNT(*)
+FROM bookings AS b
+JOIN guests AS g ON g.id = b.guest_id
+GROUP BY b.check_in, b.check_out, g.first_name, g.last_name
+HAVING COUNT(*) = 2;
+```
+c)
+```
+SELECT b.room_id, g.first_name, g.last_name, COUNT(*)
+FROM guests AS g
+JOIN bookings AS b ON b.guest_id = g.id
+GROUP BY b.room_id, g.first_name, g.last_name
+HAVING COUNT(*) > 1;
+```
+
+d)
+```
+SELECT bookings.room_id, COUNT(DISTINCT guests.first_name) AS unique_guests
+FROM bookings 
+JOIN guests ON guests.id = bookings.guest_id
+GROUP BY bookings.room_id;
+```
+2)
+
+a)
+
+**What classes/entities do we need?**
+
+We definitely need a `students` entity, as well as a `course` entity in order to track the students in the school, as well as the courses give in the school. 
+
+Since the question requires that we have also the grades for each student taking a class, we will need a `grades` entity as well. 
+
+**What fields/attributes will each entity need**
+
+A `students` entity will need the following attributes `id`, `first_name`, and `last_name`. 
+
+The `courses` entity will need the following attributes: `id`, and `course_name`. 
+
+The `grades` entity will need the following attributes: `id`, `course_id`, and `student_id`.
```

Comment:
no value?

Reply:
Did you want me to add an actual example value here for each attribute?

I didn't add anything here since I was going to talk about the types later on in this question.